lib/exercism/user_statistics.rb

```diff
@@ -0,0 +1,30 @@
+class UserStatistics
+  def self.of(username)
+    by_username(username)
+  rescue Exception => e
+    Bugsnag.notify(e, {username: username})
+    id
+  end
+
+  def self.by_username(username)
+    user = User.find_by(username: username)
+    { user: user.public_user_attributes,
+      statistics: user_problem_statistcs(user) }
+  end
+
+  def self.problem_total(track)
+    track.problems.count
+  end
+
+  def self.problem_completion(user, track)
+    user.exercises.where("LOWER(language) = ?", track.language.downcase).count
+  end
+
+  def self.user_problem_statistcs(user)
+    X::Track.all.map.with_object({}) do |track, statistics|
+      statistics[track.language] = { total: problem_total(track),
```

Comment:
Do we want the language or the track ID here? (Language: "C++", Track ID: "cpp")